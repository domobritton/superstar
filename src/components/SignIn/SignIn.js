import React from 'react';

import { STATUS } from 'gitstar-components';

import superStar from '../../superstar.svg';
import styled from 'styled-components';

export const SignIn = ({ status, CLIENT_ID, REDIRECT_URI }) => {
  return (
    <Page>
      <Logo src={superStar} alt="superstar" />
      {status === STATUS.INITIAL && (
        <Login
          href={`https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&scope=user&redirect_uri=${REDIRECT_URI}`}
        >
          LOGIN
        </Login>
      )}
    </Page>
  );
};

const Page = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100vh;
  background: gray;
`;

const Logo = styled.img`
  width: 450px;
`;

const Login = styled.a`
  color: white;
  font-size: 22px;
  &:hover {
    text-decoration-line: none;
    color: gold;
  }
`;

export default SignIn;
