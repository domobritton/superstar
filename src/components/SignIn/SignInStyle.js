import styled from 'styled-components';

export const Page = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  height: 100vh;
  background: gray;
`;

export const Logo = styled.img`
  width: 450px;
`;

export const Login = styled.a`
  color: white;
  font-size: 22px;
  &:hover {
    text-decoration-line: none;
    color: gold;
  }
`;
