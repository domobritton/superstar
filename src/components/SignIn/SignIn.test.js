import React from 'react';
import { render, cleanup } from 'react-testing-library';
import { SignIn } from './SignIn';

afterEach(cleanup);

test('<SignIn />', () => {
  const { container } = render(<SignIn />);
  expect(container.firstChild).toMatchSnapshot();
});
