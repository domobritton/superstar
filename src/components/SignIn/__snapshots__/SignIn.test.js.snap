// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignIn /> 1`] = `
<div
  class="sc-bdVaJa fQQRvV"
>
  <img
    alt="superstar"
    class="sc-bwzfXH fOkLGk"
    src="superstar.svg"
  />
</div>
`;
